src/Rating.js

```diff
@@ -1,8 +1,228 @@
 'use strict';
 
 import RatingBase from './Rating.soy';
+import { core } from 'metal';
 
 class Rating extends RatingBase {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', 'a', this.handleMouseOverEvent.bind(this));
+        this.delegate('click', 'a', this.handleClickEvent.bind(this));
+        this.on('mouseout', this.handleMouseOutEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.target.dataset.index, 10);
```

Comment:
Using `delegateTarget` is safer, since it will always be the element that matched the selector, while `target` could be some element inside it that was initially clicked.